Update overview layout

annavik opened the pull request:
## Summary

In this PR we add a sidebar layout to the project overview page, to make this page more scalable (enable us to add more views) and also to bring some structure. See related issues for more background.

### List of Changes

* Add sidebar to project overview and update routing
* Hide pagination bar for compact table views (when we only have one page of items to show)
* Make it possible to edit and delete project from overview instead of project gallery
* Setup routing for processing details and pipeline details
* Add small tweak to breadcrumbs, to better handle missing project (for example after delete)

### Related Issues

Closes #753 and #754.

### Screenshots

Before:
<img width="1728" alt="Screenshot 2025-03-19 at 18 07 32" src="https://github.com/user-attachments/assets/da1167f6-0594-4b1c-9b75-1a787395941d" />

After:
<img width="1728" alt="Screenshot 2025-03-19 at 18 07 15" src="https://github.com/user-attachments/assets/c72027bd-9fe3-42ac-bcc6-3390bde7a17c" />

### Comments
In future, I think we should add a page "General" to the section "Settings" and remove the edit and delete button from the sidebar, to better streamline how project details are edited.

Upcoming pages "Exports", "Algorithms" and "Team" could fit in like this:
<img width="311" alt="Screenshot 2025-03-19 at 17 00 53" src="https://github.com/user-attachments/assets/933b3db7-e9c1-4126-a678-fd9ab3869bd6" />

mihow reviewed (approved):
It's fabulous! What a big refactor on the code side as well. It's so great to have the new proper routes with list & detail URLs. And thanks for breaking out entities into their own files.

mihow commented:
@annavik I'm leaving this for you to merge when you are ready!

annavik commented:
Thanks @mihow, I like the name "Project"! I have pushed some updates for this change and also I decided to take the time to setup a view "General" for project details to avoid confusions! :)

### Summary
- Changed name "Overview" -> "Project"
- Updated page folder name from "overview" -> "project"
- Removed main view breadcrumb when view "Project" is active (the project name is already part of breadcrumbs so it felt a bit repetitive)
- Included new view "General" for project details settings (only visible if a user is allowed to edit)

### Screenshots
Before:
<img width="1728" alt="Screenshot 2025-03-20 at 11 04 09" src="https://github.com/user-attachments/assets/b6c00d9e-e06a-4823-bd76-63b1d3840986" />

After:
<img width="1728" alt="Screenshot 2025-03-20 at 11 05 01" src="https://github.com/user-attachments/assets/0dea53b7-f654-4fc7-80a7-957f5a89d46f" />
